### ISSUE TITLE
iRacing position properties

### ISSUE THREAD

**Issue by andreasdahl1987** [SHWotever/SimHub#631]
Hi! 

For some reason the race position of player and opponents are updated only sporadically in iRacing, and not at all during the first lap of the race. However, the relative time gap to opponents is updated continously (using the drivergaptoplayer() function). I've made a javascript looking through all the drivergaptoplayer(x) values in a for loop, and successfully returning my correct race position that updates continously. However, doing this for 30 race participants is too heavy and Simhub stutters to a halt (at least with my technique). Here is how I find out which driver is actually in 2nd position:

function pos2(){
   
    var pos = 2;

    if(root["pos2"] == null) {
        root["pos2"] = []
    }

    if(root["pos2value"] == null) {
        root["pos2value"] = 0
    }

    for (i=1;drivergaptoplayer(i) != null;i++) {
        root["pos2"].push(drivergaptoplayer(i));
    }

    var sorted = root["pos2"].sort();
    var extracted = sorted[pos];

    for (u=1;drivergaptoplayer(u) != null;u++) {
        if(drivergaptoplayer(u) == extracted) {
            root["pos2value"] = u;
        }
    }

    return root["pos2value"]

}

**So to sum it up; since simhub can correctly return the relative position in time of all the race participants on the race track, why cant it correcly return the race position for the same participants? Can this be fixed within the software? Leadersboards such as JRT has perfect real-time presentation of race position, Im sure simhub could do the same :D**

**Comment by SHWotever** [comment]
Hi ! Currently I use the results position which are the official results positions as given by the game. I will look if I can feed with another data without breaking the "official results" aspect.

**Comment by andreasdahl1987** [comment]
Thanks :) Meanwhile I've made some scripts to determine the correct position of myself (n), n+1, n-1 and leader. This is all I really needed, though it would be nice to be able to make a (perfectly) working leaderboard for iRacing with simhub :D
